lib/hexpm/short_urls/short_url.ex

```diff
@@ -0,0 +1,58 @@
+defmodule Hexpm.ShortURLs.ShortURL do
+  use Hexpm.Schema
+  alias Hexpm.ShortURLs
+  alias Hexpm.ShortURLs.ShortURL
+
+  schema "short_urls" do
+    field :url, :string
+    field :short_code, :string
+
+    timestamps()
+  end
+
+  def changeset(params) do
+    %ShortURL{}
+    |> cast(params, [:url])
+    |> validate_required([:url])
+    |> ensure_url_domain()
+    |> put_change(:short_code, generate_random(5))
+    |> validate_required(:short_code, message: "could not generate a unique short code")
```

Comment:
this would catch the short_code being blank but it's impossible since we have `put_change` above.

```suggestion
```

Reply:
this is to catch the situation where we didnt find a unique short code `defp generate_random(_length, 0), do: nil`